src/OpenTelemetry.h

```diff
@@ -0,0 +1,69 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#pragma once
+
+#include "HttpServer.h"
+#include "opentelemetry/proto/collector/metrics/v1/metrics_service.pb.h"
+#include <functional>
+#include <timer.hpp>
+
+namespace visor {
+
+constexpr char BIN_CONTENT_TYPE[] = "application/x-protobuf";
+
+using namespace opentelemetry::proto;
+
+struct OtelConfig {
+    bool enable{false};
+    std::string endpoint{"localhost"};
+    uint32_t port_number{4317};
+    uint64_t interval_sec{60};
+    std::string tls_cert;
+    std::string tls_key;
+};
+
+class OpenTelemetry
+{
+    std::unique_ptr<httplib::Client> _client;
+    collector::metrics::v1::ExportMetricsServiceRequest _request;
+    metrics::v1::ResourceMetrics *_resorce;
+    std::shared_ptr<timer::interval_handle> _timer_handle;
+    std::function<bool(metrics::v1::ResourceMetrics &resource)> _callback;
+
+public:
+    OpenTelemetry(const OtelConfig &config)
+    {
+        if (!config.tls_cert.empty() && !config.tls_key.empty()) {
+            _client = std::make_unique<httplib::Client>(config.endpoint, config.port_number, config.tls_cert, config.tls_key);
+        } else {
+            _client = std::make_unique<httplib::Client>(config.endpoint, config.port_number);
+        }
+        _resorce = _request.add_resource_metrics();
+        static timer timer_thread{std::chrono::seconds(config.interval_sec)};
+        _timer_handle = timer_thread.set_interval(std::chrono::seconds(config.interval_sec), [this] {
+            _resorce->clear_scope_metrics();
+            if (_callback && _callback(*_resorce)) {
+                if (auto body_size = _request.ByteSizeLong(); body_size > sizeof(_request)) {
+                    auto body = std::make_unique<char[]>(body_size);
+                    _request.SerializeToArray(body.get(), body_size);
+                    auto result = _client->Post("/v1/metrics", body.get(), body_size, BIN_CONTENT_TYPE);
```

Comment:
this URL doesn't need to be configurable?

Reply:
```
OTLP/HTTP Request
Telemetry data is sent via HTTP POST request. The body of the POST request is a payload either in binary-encoded Protobuf format or in JSON-encoded Protobuf format.

The default URL path for requests that carry trace data is /v1/traces (for example the full URL when connecting to "example.com" server will be https://example.com/v1/traces). The request body is a Protobuf-encoded ExportTraceServiceRequest message.

The default URL path for requests that carry metric data is /v1/metrics and the request body is a Protobuf-encoded ExportMetricsServiceRequest message.

The default URL path for requests that carry log data is /v1/logs and the request body is a Protobuf-encoded ExportLogsServiceRequest message.

The client MAY gzip the content and in that case MUST include "Content-Encoding: gzip" request header. The client MAY include "Accept-Encoding: gzip" request header if it can receive gzip-encoded responses.

Non-default URL paths for requests MAY be configured on the client and server sides.
```
yeah, I think it should be configurable as well

Reply:
I make it configurable